### DISCUSSION TITLE
Magentic One - how to get the formatted output as in the published videos from the blog?

### DISCUSSION THREAD

**Discussion by MSDNAndi** [microsoft/autogen#4251] [Q&A] [answered]
Hello together,

has anybody seen in the repo how to run Magentic One to get the formatted output as it is seen in the videos published at the Blog post
https://www.microsoft.com/en-us/research/articles/magentic-one-a-generalist-multi-agent-system-for-solving-complex-tasks/

Kind regards,

**Comment by husseinmozannar** [comment, marked as answer]
Hi,

We created a temporary log viewer as a flask app, you point it to the directory where you stored the logs from Magentic-One (including the screenshots from the websurfer), and it replays the logs. It should be possible to make it work live. 

This is not a maintained file and is just temporary, soon AutoGen Studio will have a similar & better interface to interact with Magentic-One

[log_viewer, not maintened](https://github.com/microsoft/autogen/blob/e46bac503449c7c4b083110918baa55767d97d56/python/packages/autogen-magentic-one/interface/log_viewer.py)

You need to install

`pip install flask markdown
`

To run the log viewer, use the following command:

```
python log_viewer.py <log_folder> --port <port>
```
Where:

<log_folder>: Path to the folder containing JSONL log files.
: Port number to run the server on (default: 5000).
The log viewer provides a web interface to view logs in the browser. It will start by default on http://localhost:5000/.

**Reply by MSDNAndi** [reply]
Thank you for the response. I manually got the file and the icons it needs.
It seems to work, that said, it doesn't seem to work live for me, doesn't pick up new events once it reaches the end of the file. The code implies it tries to reload the logs every 5 seconds though. 
A "pause" and then resuming will continue until it reaches the end again.
I am running it in WSL2 (Ubuntu).

**Reply by husseinmozannar** [reply]
You're right, sorry I should have been clearer, I meant to imply that with some code changes it can be made live, right now that bug exists. We're working on a port to AutoGen Studio for a better experience.
